tasks/sep/ink-001-fp-holocene-pectra-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,233 @@
+
+        require(faultDisputeGame.version().eq(gameVersion), "game-100");
+        require(permissionedDisputeGame.version().eq(gameVersion), "game-200");
+
+        require(faultDisputeGame.absolutePrestate().raw() == absolutePrestate, "game-300");
+        require(permissionedDisputeGame.absolutePrestate().raw() == absolutePrestate, "game-400");
+
+        require(address(faultDisputeGame.vm()) == newMips, "game-500");
+        require(address(permissionedDisputeGame.vm()) == newMips, "game-600");
+
+        require(faultDisputeGame.l2ChainId() == chainId, "game-700");
+        require(permissionedDisputeGame.l2ChainId() == chainId, "game-800");
+    }
+
+    /// @notice Checks that the FaultDisputeGame was handled correctly.
+    function checkFaultDisputeGame() internal view {
+        // Check that the FaultDisputeGame version is correct.
+        require(LibString.eq(expectedFaultDisputeGameImpl.version(), "1.3.1"), "checkFaultDisputeGame-21");
+
+        // Check that only bytecode diffs vs comparison contract are expected.
+        BytecodeComparison.Diff[] memory diffs = new BytecodeComparison.Diff[](12);
+        diffs[0] = BytecodeComparison.Diff({start: 1319, content: abi.encode(expectedDelayedWETHProxy)});
+        diffs[1] = BytecodeComparison.Diff({start: 1555, content: abi.encode(proxies.AnchorStateRegistry)});
+        diffs[2] = BytecodeComparison.Diff({start: 1926, content: abi.encode(absolutePrestate)});
+        diffs[3] = BytecodeComparison.Diff({start: 2590, content: abi.encode(chainConfig.chainId)});
+        diffs[4] = BytecodeComparison.Diff({start: 6026, content: abi.encode(proxies.AnchorStateRegistry)});
+        diffs[5] = BytecodeComparison.Diff({start: 6476, content: abi.encode(expectedDelayedWETHProxy)});
+        diffs[6] = BytecodeComparison.Diff({start: 9344, content: abi.encode(expectedDelayedWETHProxy)});
+        diffs[7] = BytecodeComparison.Diff({start: 9704, content: abi.encode(proxies.AnchorStateRegistry)});
+        diffs[8] = BytecodeComparison.Diff({start: 10730, content: abi.encode(expectedDelayedWETHProxy)});
+        diffs[9] = BytecodeComparison.Diff({start: 12628, content: abi.encode(absolutePrestate)});
+        diffs[10] = BytecodeComparison.Diff({start: 14649, content: abi.encode(chainConfig.chainId)});
+        diffs[11] = BytecodeComparison.Diff({start: 15892, content: abi.encode(expectedDelayedWETHProxy)});
+
+        require(
+            BytecodeComparison.compare(
+                address(comparisonFaultDisputeGameImpl), address(expectedFaultDisputeGameImpl), diffs
+            ),
+            "checkFaultDisputeGame-101"
+        );
+    }
+
+    /// @notice Checks that the PermissionedDisputeGame was handled correctly.
+    function checkPermissionedDisputeGame() internal view {
+        // Check that the PermissionedDisputeGame version is correct.
+        require(LibString.eq(expectedPermissionedDisputeGameImpl.version(), "1.3.1"), "checkPermissionedDisputeGame-20");
```

Comment:
Same as above: can we also check the addresses of the `DelayedWETH` and `AnchorStateRegistry` references?